utils/buffer.py

```diff
@@ -278,15 +283,19 @@ def get_class_sample_count(self, label: int) -> torch.Tensor:
         else:
             return 0
 
-    def flush_class(self, label: int) -> None:
+    def flush_class(self, label: int, metaclass: bool = False) -> None:
         """
         Removes all samples with given label.
         If label not present in the buffer, then raise ValueError exception.
         """
-        idx = torch.argwhere(self.labels != label).flatten()
-        num_removed = len(self.labels) - len(idx)
+        if metaclass:
+            idx = torch.argwhere(self.original_labels != label).flatten()
+            num_removed = len(self.original_labels) - len(idx)
+        else:
+            idx = torch.argwhere(self.labels != label).flatten()
+            num_removed = len(self.labels) - len(idx)
```

Comment:
the name of the argument metaclass is misleading.
if metaclass is true you return original labels (fine-grained), otherwise you return meta-labels (metaclass). 
Please use other name for that variable,